src/app_manager/app_manager.py

```diff
@@ -214,18 +283,69 @@ def handle_start_app(self, req):
             rospy.loginfo("Launching: %s"%(app.launch))
             self._status_pub.publish(AppStatus(AppStatus.INFO, 'launching %s'%(app.display_name)))
 
+            plugin_launch_files = []
+            if app.plugins:
+                self._current_plugins = []
+                for app_plugin in app.plugins:
+                    app_plugin_type = app_plugin['type']
+                    plugin = [p for p in self._plugins if p['name'] == app_plugin_type][0]
```

Comment:
```suggestion
                    plugin = next(p for p in self._plugins if p['name'] == app_plugin_type)
```

you probably also want to handle the `StopIteration` exception if the user mistyped and provide a useful error message.